### ISSUE TITLE
Modify BottleNeck with DCN but Precision and Recall are 0

### ISSUE THREAD

**Issue by Steinwang** [ultralytics/yolov5#5831]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I modified BottleNeck module with Deformable ConvNets, got similiar results, the loss goes down normally, but on validation set the Network can't find any objects, the Precision and Recall remains 0 during the whole training procedure, does any one know how to fix it?

**Comment by Steinwang** [comment]
On my custom datasets

**Comment by glenn-jocher** [comment]
@Steinwang 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by Zengyf-CVer** [comment]
@Steinwang 
Can you send me an unusual screenshot of your training? In the shell.
I have encountered this situation before, and I can estimate from your description that your training epochs are too few.
In YOLOv5, this custom model architecture will cause the mAP to appear 0 in the initial epochs, so you can try to increase epochs, such as `--epochs 200`, and wait patiently.

**Comment by Steinwang** [comment]
![image](https://user-images.githubusercontent.com/47937124/144174444-88572542-b76f-4b82-a03f-8a422d171d01.png)
like this，during the whole training it  cannot find any object on validation set and test set @Zengyf-CVer

**Comment by Zengyf-CVer** [comment]
@Steinwang 
The problem occurs when your `label` is 0. It means that there is a problem with your validation set label. YOLOv5 will detect your tags when it starts running. You can check the prompts at the top of this screenshot.
![1111111](https://user-images.githubusercontent.com/41098760/144179001-1039285d-b191-4c47-8d6f-309890cfb874.png)

**Comment by Steinwang** [comment]
> 
I random divide the train set to 90% for training and 10% for validation at the start of training, do you mean I should delete the train.cache and val.cache every time before training starts? By the way, is it possible we talk in Chinese?  @Zengyf-CVer

**Comment by Zengyf-CVer** [comment]
@Steinwang 
There is no need to delete the cache. There is a problem with your verification set labels, and the program cannot find your verification set labels, so it is 0.

**Comment by Steinwang** [comment]
> @Zengyf-CVer 
So this is not due to the Network, it is beacuse the labels I divided?

**Comment by Zengyf-CVer** [comment]
@Steinwang 
Yes, carefully check your validation set label, whether it meets the YOLOv5 training standard.

**Comment by yustaub** [comment]
hello, sir. Does DCN useful in your training result? @Steinwang
